core/receipt.go

```diff
@@ -62,12 +65,33 @@ func messagesSentHash(messages []*L2ToL1Message) *felt.Felt {
 
 func receiptCommitment(receipts []*TransactionReceipt) (*felt.Felt, error) {
 	var commitment *felt.Felt
-
 	return commitment, trie.RunOnTempTriePoseidon(commitmentTrieHeight, func(trie *trie.Trie) error {
-		for i, receipt := range receipts {
-			receiptTrieKey := new(felt.Felt).SetUint64(uint64(i))
-			_, err := trie.Put(receiptTrieKey, receipt.hash())
-			if err != nil {
+		numWorkers := min(runtime.GOMAXPROCS(0), len(receipts))
+		results := make([]*felt.Felt, len(receipts))
+		wg := sync.WaitGroup{}
+		wg.Add(numWorkers)
+
+		jobs := make(chan int, len(receipts))
+		for i := range receipts {
+			jobs <- i
+		}
+		close(jobs)
+
+		for range numWorkers {
+			go func() {
+				defer wg.Done()
+				for i := range jobs {
+					results[i] = receipts[i].hash()
+				}
+			}()
+		}
+
+		wg.Wait()
+
+		key := new(felt.Felt)
```

Comment:
just double checking: are you sure that `trie.Put(...)` don't mess with key pointer ?

Reply:
I am sure, but I think it would be safer to change.